core/station/impl/src/models/request_policy_rule.rs

```diff
@@ -44,6 +45,77 @@ pub enum RequestPolicyRule {
     NamedRule(NamedRuleId),
 }
 
+// Implement Display with circular reference detection for NamedRules
+impl fmt::Display for RequestPolicyRule {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut visited = HashSet::new();
+        self.fmt_with_context(f, &mut visited)
+    }
+}
+
+impl RequestPolicyRule {
+    fn fmt_with_context(
+        &self,
+        f: &mut fmt::Formatter<'_>,
+        visited: &mut HashSet<NamedRuleId>,
+    ) -> fmt::Result {
+        match self {
+            RequestPolicyRule::AutoApproved => write!(f, "AutoApproved"),
+            RequestPolicyRule::QuorumPercentage(_, _) => write!(f, "QuorumPercentage"),
+            RequestPolicyRule::Quorum(_, _) => write!(f, "Quorum"),
+            RequestPolicyRule::AllowListedByMetadata(_) => write!(f, "AllowListedByMetadata"),
+            RequestPolicyRule::AllowListed => write!(f, "AllowListed"),
+            RequestPolicyRule::Or(rules) => {
+                write!(f, "Or(")?;
+                for (i, rule) in rules.iter().enumerate() {
+                    if i > 0 {
+                        write!(f, ",")?;
+                    }
+                    rule.fmt_with_context(f, visited)?;
+                }
+                write!(f, ")")
+            }
+            RequestPolicyRule::And(rules) => {
+                write!(f, "And(")?;
+                for (i, rule) in rules.iter().enumerate() {
+                    if i > 0 {
+                        write!(f, ",")?;
+                    }
+                    rule.fmt_with_context(f, visited)?;
+                }
+                write!(f, ")")
+            }
+            RequestPolicyRule::Not(rule) => {
+                write!(f, "Not(")?;
+                rule.fmt_with_context(f, visited)?;
+                write!(f, ")")
+            }
+            RequestPolicyRule::NamedRule(id) => {
+                if !visited.insert(*id) {
+                    return write!(
+                        f,
+                        "NamedRule(CIRCULAR_REFERENCE {})",
+                        Uuid::from_bytes(*id).hyphenated()
```

Comment:
I'm not sure if the id is useful here since we don't print the matching id in the parent.

Reply:
My suggestion: only print `"NamedRule(CIRCULAR_REFERENCE)"`.